pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +101,202 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
 	for _, warn := range warns {
 		result.Add(errors.WarnInvalidCSV(warn.Error(), bundle.CSV.GetName()))
 	}
+
+	errs, warns = validateHubDeprecatedAPIS(bundle, k8sVersion)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	if warn := validateHubChannels(bundle.Channels); warn != nil {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
 	return result
 }
 
-func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+// validateHubChannels will check the channels. The motivation for the following check is to ensure that operators
+// authors knows if their operator bundles are or not respecting the Naming Convention Rules.
+// However, the operator authors still able to choose the names as please them.
+func validateHubChannels(channels []string) error {
+	const preview = "preview"
+	const stable = "stable"
+	const fast = "fast"
+	for _, channel := range channels {
+		if !strings.Contains(channel, preview) &&
+			!strings.Contains(channel, stable) &&
+			!strings.Contains(channel, fast) {
```

Comment:
So package maintainers can only use channel strings that contain one of the above 3? Why can't a maintainer have these 3 plus other arbitrary channel names?

Reply:
@estroz it is clarified in the comment of this check. These are the naming suggested "convention". See that will raise a warning and will share the link: https://olm.operatorframework.io/docs/best-practices/channel-naming/ to ensure that they know about that. However, they are still able to choose the names which please them.

Reply:
Right, but I should be able to define any channel name I want as long as preview/fast/stable are also defined for a bundle. This check really should be
```go
for _, conventionalChannelPrefix := range []string{preview, fast, stable} {
	if !containsPrefix(channels, conventionalChannelPrefix) {
		errs = append(errs, fmt.Errorf("bundle channels should contain a %s channel", conventionalChannel))
	}
}
```